iothub_client/samples/iothub_client_device_cbor_sample/iothub_client_device_cbor_sample.c

```diff
@@ -0,0 +1,956 @@
+    }
+    (void)printf("\n");
+
+    // Publish the telemetry message. Set callback for telemetry send confirmation.
+    uint64_t id = telemetry_message_id;
+    IoTHubDeviceClient_SendEventAsync(iothub_client_handle, message_handle, send_telemetry_confirm_callback, &id);
+    (void)printf("The device client published the telemetry message. telemetry_message_id: %" PRIu64 ".\n\n", id);
+
+    // The message is copied to the SDK so we can destroy it.
+    IoTHubMessage_Destroy(message_handle);
+    telemetry_message_id++; // Increase for next telemetry message.
+}
+
+//
+// Callbacks
+//
+static void connection_status_callback(IOTHUB_CLIENT_CONNECTION_STATUS result, IOTHUB_CLIENT_CONNECTION_STATUS_REASON reason, void* user_context)
+{
+    (void)reason;
+    (void)user_context;
+
+    if (result == IOTHUB_CLIENT_CONNECTION_AUTHENTICATED)
+    {
+        (void)printf("\nconnection_status_callback: The device client is connected to an IoT Hub.\n\n");
+    }
+    else
+    {
+        (void)printf("\nconnection_status_callback: The device client has been disconnected.\n\n");
+    }
+}
+
+static void get_twin_async_callback(DEVICE_TWIN_UPDATE_STATE update_state, const unsigned char* payload, size_t size, void* userContextCallback)
+{
+    (void)userContextCallback;
+
+    (void)printf("\nget_twin_async_callback payload: ");
+    for (size_t i = 0; i < size; ++i)
+    {
+        (void)printf("%02X ", payload[i]);
+    }
+    (void)printf("\n");
+
+    // Parse the twin desired properties received from IoT Hub.
+    Car iot_hub_car;
+    memset(&iot_hub_car, 0, sizeof(Car));
+    parse_cbor_desired_property(update_state, payload, &iot_hub_car);
+
+    // Update device_car from twin desired properties received from IoT Hub.
+    // change_oil_reminder
+    {
+        bool temp_value = device_car.change_oil_reminder;
+        device_car.change_oil_reminder = iot_hub_car.change_oil_reminder;
+        (void)printf("The device client updated `%s` locally from %s to %s.\n", twin_property_change_oil_reminder_name, temp_value ? "true" : "false", device_car.change_oil_reminder ? "true" : "false");
+    }
+
+    // state.max_speed
+    if ((iot_hub_car.state.max_speed != 0))
+    {
+        if (iot_hub_car.state.max_speed > 260)
+        {
+            (void)printf("Desired `max_speed` of %" PRIu64 " exceeds device capability. Rejecting update.\n", iot_hub_car.state.max_speed);
+        }
+        else
+        {
+            uint64_t temp_value = device_car.state.max_speed;
+            device_car.state.max_speed = iot_hub_car.state.max_speed;
+            (void)printf("The device client updated `%s` locally from %" PRIu64 " to %" PRIu64 ".\n", twin_property_state_max_speed_name, temp_value, iot_hub_car.state.max_speed);
+        }
+    }
+    else
+    {
+        (void)printf("`max_speed` cannot be 0. Rejecting update.\n");
+    }
+
+    // state.software_version
+    if (iot_hub_car.state.software_version != 0)
```

Comment:
for doubles please use something similar to:

```c
#define SAMPLE_FLOAT_PRECISION 1e-2
if( fabs(iot_hub_car.state.software_version < SAMPLE_FLOAT_PRECISION) );
```

Reply:
I'm not sure I'm following this. This shows that the software version can be in the range (-0.01, 0.01) ?  I think I will just check that it is positive.  It an always go more detailed but it's just a sample.